contracts/contracts/validator-registry/ValidatorRegistryV1.sol

```diff
@@ -174,39 +183,55 @@ contract ValidatorRegistryV1 is IValidatorRegistryV1, ValidatorRegistryV1Storage
+            action(pubKey, stakeAmount, withdrawalAddress);
+        }
+    }
+
     /*
      * @dev Internal function to stake ETH on behalf of one or multiple validators via their BLS pubkey.
      * @param blsPubKeys The validator BLS public keys to stake.
      * @param withdrawalAddress The address to receive the staked ETH.
      */
     function _stake(bytes[] calldata blsPubKeys, address withdrawalAddress) internal {
-        require(blsPubKeys.length > 0, "There must be at least one recipient");
-        uint256 splitAmount = msg.value / blsPubKeys.length;
-        for (uint256 i = 0; i < blsPubKeys.length; i++) {
-            bytes calldata pubKey = blsPubKeys[i];
-            stakedValidators[pubKey] = StakedValidator({
-                exists: true,
-                balance: splitAmount,
-                withdrawalAddress: withdrawalAddress,
-                unstakeHeight: EventHeightLib.EventHeight({ exists: false, blockHeight: 0 })
-            });
-            emit Staked(msg.sender, withdrawalAddress, pubKey, splitAmount);
-        }
+        _splitStakeAndApplyAction(blsPubKeys, withdrawalAddress, _stakeAction);
+    }
+
+    /// @dev Internal function that creates a staked validator record and emits a Staked event.
+    function _stakeAction(bytes calldata pubKey, uint256 stakeAmount, address withdrawalAddress) internal {
+        stakedValidators[pubKey] = StakedValidator({
+            exists: true,
+            balance: stakeAmount,
+            withdrawalAddress: withdrawalAddress,
+            unstakeHeight: EventHeightLib.EventHeight({ exists: false, blockHeight: 0 })
+        });
+        emit Staked(msg.sender, withdrawalAddress, pubKey, stakeAmount);
     }
 
     /* 
      * @dev Internal function to add ETH to the staked balance of one or multiple validators via their BLS pubkey.
      * @param blsPubKeys The BLS public keys to add stake to.
      */
     function _addStake(bytes[] calldata blsPubKeys) internal {
-        require(blsPubKeys.length > 0, "There must be at least one recipient");
-        uint256 splitAmount = msg.value / blsPubKeys.length;
-        for (uint256 i = 0; i < blsPubKeys.length; i++) {
-            bytes calldata pubKey = blsPubKeys[i];
-            stakedValidators[pubKey].balance += splitAmount;
-            emit StakeAdded(msg.sender, stakedValidators[pubKey].withdrawalAddress,
-                pubKey, splitAmount, stakedValidators[pubKey].balance);
-        }
+        _splitStakeAndApplyAction(blsPubKeys, address(0), _addStakeAction);
+    }
+
+    /// @dev Internal function that adds stake to an already existing validator record, emitting a StakeAdded event.
+    function _addStakeAction(bytes calldata pubKey, uint256 stakeAmount, address) internal {
+        stakedValidators[pubKey].balance += stakeAmount;
+        emit StakeAdded(msg.sender, stakedValidators[pubKey].withdrawalAddress, pubKey, stakeAmount, stakedValidators[pubKey].balance);
     }
```

Comment:
```solidity
function _addStakeAction(bytes calldata pubKey, uint256 stakeAmount, address) internal {
    Validator storage validator = stakedValidators[pubKey];
    validator.balance += stakeAmount;
    emit StakeAdded(msg.sender, validator.withdrawalAddress, pubKey, stakeAmount, validator.balance);
}
```

By caching the stakedValidators[pubKey] in a local variable (validator), we reduce the number of storage reads